test/metabase/driver/postgres_test.clj

```diff
@@ -1321,40 +1321,46 @@
                                      "CREATE VIEW \"dotted.schema\".\"dotted.view\" AS SELECT 'hello world';"
                                      "CREATE MATERIALIZED VIEW \"dotted.schema\".\"dotted.materialized_view\" AS SELECT 'hello world';"
                                      "DROP ROLE IF EXISTS privilege_rows_test_example_role;"
                                      "CREATE ROLE privilege_rows_test_example_role WITH LOGIN;"
-                                     "GRANT SELECT (id) ON \"dotted.schema\".\"dotted.table\" TO privilege_rows_test_example_role;"
+                                     "GRANT SELECT ON \"dotted.schema\".\"dotted.table\" TO privilege_rows_test_example_role;"
                                      "GRANT UPDATE ON \"dotted.schema\".\"dotted.table\" TO privilege_rows_test_example_role;"
+                                     "GRANT SELECT (id) ON \"dotted.schema\".\"dotted.partial_select\" TO privilege_rows_test_example_role;"
+                                     "GRANT UPDATE (id) ON \"dotted.schema\".\"dotted.partial_update\" TO privilege_rows_test_example_role;"
+                                     "GRANT INSERT (text) ON \"dotted.schema\".\"dotted.partial_insert\" TO privilege_rows_test_example_role;"
                                      "GRANT SELECT ON \"dotted.schema\".\"dotted.view\" TO privilege_rows_test_example_role;"
                                      "GRANT SELECT ON \"dotted.schema\".\"dotted.materialized_view\" TO privilege_rows_test_example_role;"))
            (testing "check that without USAGE privileges on the schema, nothing is returned"
              (is (= #{}
                     (get-privileges))))
            (testing "with USAGE privileges, SELECT and UPDATE privileges are returned"
              (jdbc/execute! conn-spec "GRANT USAGE ON SCHEMA \"dotted.schema\" TO privilege_rows_test_example_role;")
-             (is (= #{{:role   nil
-                       :schema "dotted.schema"
-                       :table  "dotted.materialized_view"
-                       :update false
-                       :select true
-                       :insert false
-                       :delete false}
-                      {:role   nil
-                       :schema "dotted.schema"
-                       :table  "dotted.view"
-                       :update false
-                       :select true
-                       :insert false
-                       :delete false}
-                      {:role   nil
-                       :schema "dotted.schema"
-                       :table  "dotted.table"
-                       :select true
-                       :update true
-                       :insert false
-                       :delete false}}
+             (is (= (into #{}
+                          (map #(merge {:role   nil
+                                        :schema "dotted.schema"
+                                        :table  "dotted.materialized_view"
```

Comment:
I'd rather exclude this default table name, could cause a lot of head scratching at test output if we forgot to override it.

Reply:
Also nit, I think it'd be a little bit more obvious what's going on if you bound this map to something like `base-privileges`.

Reply:
> I think it'd be a little bit more obvious what's going on if you bound this map to something like base-privileges

I agree with that if we reuse this function elsewhere, but in this context I like that I could see the default value in a single map opreration

Reply:
It took my eyes a second to realise what's going on - checking which way around the merge is happening, then inferring the purpose of the merge. It wasn't helped by the `:table` key in both though.

Oh one more nit is that you could use `(set ...)` instead of `(into #{} ...)`.